### PR TITLE
#160710730 Adds reason and duration for blocking room for maintenance and the duration it's going to take.

### PR DESCRIPTION
**Pull request by dmaish**
#### What does this PR do?
Adds an endpoint to add the maintenance reason to the database.

#### Description of Task to be completed?
- Add an endpoint to add maintenance reason to the database
- Add tests for the new endpoint.

#### How should this be manually tested?
- Clone the repository and enter the project's folder
- Checkout to `ft-add-maintainance-reason-#160710730`
- Assign yourself the travel admin role by changing your roleId to 29187 in the users table
- Start the application using 'make start'
- Open postman and provide your token in the authorization header.
- If you do not have a guest house id, see [this](https://github.com/andela/travel_tool_back/pull/65) for guesthouse structure to create one
- Make a POST request to `/api/v1/room/:room-id/maintainance`. Ensure to add the `id` as the ID of any room.
`THE BODY STRUCTURE`
{
	"reason":"Broken windows",
	"start":"10/28/2018",
	"end":"2018-10-16"
}

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160710730](https://www.pivotaltracker.com/story/show/160710730)

#### Screenshots (if appropriate)
screen shot 2018-10-14 at 13 49 33

#### Questions:
None

### CODE REVIEW

**Comment by samuelkubai** [comment]
@dmaish please rename this PR title in accordance to the playbook.

**Comment by dmaish** [comment]
@ronaldndirangu , that's been fixed.

File: src/modules/guestHouse/__tests__/updateRoomStatus.test.js~HEAD [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import request from 'supertest';
```
Comment (loicemeyo) [inline src/modules/guestHouse/__tests__/updateRoomStatus.test.js~HEAD]: @dmaish is this file necessary

Comment (davenjeru) [reply src/modules/guestHouse/__tests__/updateRoomStatus.test.js~HEAD]: There is another file with the same name but without the ~HEAD. This file should be deleted @dmaish

**Comment by dmaish** [comment]
@loicemeyo , it's true that isn't the correct spelling.But I don't think it would really deter following the logic of the work as it has been consistently used in the PR.

**Comment by ezrqnkemboi** [comment]
@dmaish the work is good. I am giving some feedback here. 
```The validation of date is not work. For example, if i put start date to 12/12/2018 and put end date to 11/12/2018, there is success```
```If I put wrong format for dates in end date e.g 2018/12/01, I get this data ```:
start date must be formatted correcty mm/dd/yyyy
I think it should be end date must be formatted correctly.

**Comment by loicemeyo** [comment]
> @loicemeyo , it's true that isn't the correct spelling.But I don't think it would really deter following the logic of the work as it has been consistently used in the PR.

OK
